File: Back-restaurante/src/entities/Factura.js
const { EntitySchema } = require("typeorm");

module.exports = new EntitySchema({
  name: "Factura",
  tableName: "factura",
  columns: {
    id_factura: {
      primary: true,
      type: "number",
      generated: true,
    },
    id_cliente_fk: {
      type: "number",
      nullable: false,
    },
    id_empleado_fk: {
      type: "number",
      nullable: false,
    },
    fecha: {
      type: "date",
      nullable: true,
    },
    estado: {
      type: "varchar2",
      length: 10,
      nullable: true,
    },
  },
  relations: {
    cliente: {
      type: "many-to-one",
      target: "Cliente",
      joinColumn: {
        name: "id_cliente_fk",
        referencedColumnName: "id_cliente"
      }
    },
    empleado: {
      type: "many-to-one",
      target: "Usuario",
      joinColumn: {
        name: "id_empleado_fk",
        referencedColumnName: "id_empleado"
      }
    },
    detalles: {
      type: "one-to-many",
      target: "DetalleFactura",
      inverseSide: "factura"
    }
  }
});

File: backend/backend/src/main/java/restaurante/backend/Factura/domain/Factura.java
package restaurante.backend.Factura.domain;

import jakarta.persistence.*;
import lombok.Data;
import restaurante.backend.Cliente.domain.Cliente;
import restaurante.backend.Empleado.domain.Empleado;
import java.util.Date;

@Entity
@Table(name = "factura")
@Data
public class Factura {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id_factura;

    @ManyToOne
    @JoinColumn(name = "id_cliente_fk", nullable = false)
    private Cliente cliente;

    @ManyToOne
    @JoinColumn(name = "id_empleado_fk", nullable = false)
    private Empleado empleado;

    private Date fecha;
    private String estado;
}
